src/helper/helper.js

```diff
@@ -69,13 +69,33 @@ class DataLayerHelper {
    * Creates a new helper object for the given dataLayer.
    *
    * @param {!Array<*>} dataLayer The dataLayer to help with.
-   * @param {function(!Object<*>, !Object<*>)=} listener The callback
-   *     function to execute when a new state gets pushed onto the dataLayer.
-   * @param {boolean=} listenToPast If true, the given listener will be
-   *     executed for state changes that have already happened.
+   * @param {{listener: function(!Object<*>, !Object<*>),
+   *     listenToPast: boolean,
+   *     processNow: boolean,
+   *     commandProcessors: !Object<string,
+   *         !Array<function(*):(!Object|undefined)>>,}=} optionsObject
    */
-  constructor(dataLayer, listener = () => {
-  }, listenToPast = false) {
+  constructor(dataLayer, {
+    // The callback function to execute when a new state
+    // gets pushed onto the dataLayer.
+    'listener': listener = (() => {}),
+    // If true, the given listener will be
+    // executed for state changes that have already happened.
+    'listenToPast': listenToPast = false,
+    // If false, halts the setup of the helper and delays processing the
+    // dataLayer until the 'process' method is called. Primarily used to
+    // register processors after construction of the helper.
+    'processNow': processNow = true,
+    'commandProcessors': commandProcessors = {},
+  } = {}) {
+    // Support backwards compatibility by checking if listener
+    // function was passed in instead of options object.
+    // If so, override the listener and listenToPast legacy arguments.
+    if (typeof arguments[1] === 'function') {
+      listener = arguments[1];
+      listenToPast = arguments[2] || false;
+    }
```

Comment:
You could add a warning to the debug configuration added in #51 that says the current usage is depreciated.